### DISCUSSION TITLE
Can you use a pinia store from a Nuxt 3 server route? (defineEventHandler)

### DISCUSSION THREAD

**Discussion by jonesisg1** [vuejs/pinia#2466] [Help and Questions] [answered]
Is it possible to get the pinia instance from within defineEventHandler so that I can access my store from an api route?

```
-- server/api/mytable.patch.ts
import { useGlobalStore } from '~/stores/globals'
export default defineEventHandler(async (event) => {
  const store = useGlobalStore(???need pinia instance here???);
```

Many thanks if anyone is able to answer this.

**Comment by posva** [comment, marked as answer]
You can't but you also shouldn't need it. Server Routes are for API requests, they happen without the Nuxt (and vue) application running, so there isn't pinia, vue-router or anything like that.
Instead, call the API from the store if you want and update the values in your store as a result of the API call

**Reply by jonesisg1** [reply]
Thanks for your answer.  I realised shortly after asking the question that I was trying to do the wrong thing!  I wanted to access the user details from my API and thought I could use the store to retrieve them, but then I realised that the correct way was to pass them to the API via a Bearer token in the Authorization header.
